### PR TITLE
Enum codecs

### PR DESCRIPTION
**Pull request by Harrm**
Introduce enum codecs, apply clang-format and rename .data() in EncoderStream to .to_vector(), because it returns a temporary vector.

### CODE REVIEW

File: test/scale_optional_test.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ using scale::EncodeError;
 using scale::ScaleDecoderStream;
 using scale::ScaleEncoderStream;
 
-// TODO(yuraz): PRE-119 refactor to parameterized tests
+// TODO(Harrm): PRE-119 refactor to parameterized tests
```
Comment (xDimon) [inline test/scale_optional_test.cpp:19]: Issue label looks weird

**Review by xDimon — APPROVED** [review]
Good job!

File: test/scale_optional_test.cpp (hunk 1 above)
Comment (Harrm) [reply test/scale_optional_test.cpp:19]: Oh, it's from the good ol' times Kagome wasn't Kagome, but was simply Polkadot Runtime Environment (PRE)...

**Review by xDimon — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -8,11 +8,11 @@ cmake_minimum_required(VERSION 3.12)
 include(${CMAKE_CURRENT_LIST_DIR}/cmake/HunterGate.cmake)
 set(HUNTER_STATUS_DEBUG ON)
 HunterGate(
-        URL  https://github.com/soramitsu/soramitsu-hunter/archive/9ca72322e8d9de70d360dc7f371b223d32999123.zip
-        SHA1 37cc1150526fb9c5dcf197f918e5d9aab611823f
+    URL https://github.com/soramitsu/soramitsu-hunter/archive/9ca72322e8d9de70d360dc7f371b223d32999123.zip
```
Comment (sanblch) [inline CMakeLists.txt:11]: not versioned hunter is used

File: include/scale/detail/fixed_width_integer.hpp [hunk 3]
```diff
@@ -13,108 +13,94 @@
...
-  // static_cast to smaller size cuts them off
-  T sv = -static_cast<I>((~v) + 1);
-
-  return sv;
-}
-} // namespace scale::detail
+}  // namespace scale::detail
 
-#endif // SCALE_SCALE_UTIL_HPP
+#endif  // SCALE_SCALE_UTIL_HPP
```
Comment (sanblch) [inline include/scale/detail/fixed_width_integer.hpp:106]: extra space

**Review by sanblch — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (Harrm) [reply CMakeLists.txt:11]: Right, after the PR in Kagome is reviewed I will merge this commit in hunter in master and replace the hash here with a tag.

File: include/scale/enum_traits.hpp [hunk 4]
```diff
@@ -29,7 +32,7 @@ namespace scale {
     static constexpr bool is_default = true;
   };
 
-#define SCALE_DEFINE_ENUM_RANGE(enum_namespace, enum_name, min, max)        \
+#define SCALE_DEFINE_ENUM_VALUE_RANGE(enum_namespace, enum_name, min, max)        \
```
Comment (xDimon) [inline include/scale/enum_traits.hpp:35]: Looks as unformatted

**Review by xDimon — APPROVED** [review]

**Pull request merged**